mypy/config_parser.py

```diff
@@ -177,21 +179,38 @@ def parse_section(prefix: str, template: Options,
                               file=stderr)
                     continue
                 if key.startswith('x_'):
-                    continue  # Don't complain about `x_blah` flags
+                    pass  # Don't complain about `x_blah` flags
+                elif key.startswith('no_') and hasattr(template, key[3:]):
+                    options_key = key[3:]
+                    invert = True
+                elif key.startswith('allow') and hasattr(template, 'dis' + key):
+                    options_key = 'dis' + key
+                    invert = True
+                elif key.startswith('disallow') and hasattr(template, key[3:]):
+                    options_key = key[3:]
+                    invert = True
                 elif key == 'strict':
                     print("%sStrict mode is not supported in configuration files: specify "
                           "individual flags instead (see 'mypy -h' for the list of flags enabled "
                           "in strict mode)" % prefix, file=stderr)
                 else:
                     print("%sUnrecognized option: %s = %s" % (prefix, key, section[key]),
                           file=stderr)
-                continue
+                if invert:
+                    dv = getattr(template, options_key, None)
+                else:
+                    continue
             ct = type(dv)
         v = None  # type: Any
         try:
             if ct is bool:
                 v = section.getboolean(key)  # type: ignore  # Until better stub
+                if invert:
+                    v = not v
             elif callable(ct):
+                if invert:
+                    print("%sCan not invert non-boolean key %s" % (prefix, options_key), file=stderr)
```

Comment:
Should this kind of errors be `sys.exit(2)`?

Reply:
1. Probably
2. Similar errors are not

Reply:
@gvanrossum What do you think?

Reply:
When the mypy.ini file was new, I decided to make errors in the config file just print warnings. I can't recall why, possibly because the control flow to make them real errors was complex, or possibly because I was worried that people would have junk in their config files.

I'd stick to this tradition for now and do a separate sweep to clean it up later. (Maybe sections we don't recognize should not be fatal but in recognized sections, flags we don't understand should be fatal?)

Reply:
> I'd stick to this tradition for now and do a separate sweep to clean it up later.

SGTM.